src/uri/UriValidator.cpp

```diff
@@ -0,0 +1,159 @@
+using namespace uprotocol::v1;
+
+/**
+ * isValidUri a {@link uri} to ensure that it has at least a name for the
+ * uEntity.
+ *
+ * @param uri {@link uri} to validate.
+ * @return Returns bool containing a true or a false
+ */
+bool UriValidator::isValidUri(const std::string& uri) {
+    auto uri_view = LongUriSerializer::deserialize(uri);
+    return !isEmpty(uri_view);   
+}
+
+/**
+ * Validate a {@link UUri} to ensure that it has at least a name for the uEntity.
+ *
+ * @param uri {@link UUri} to validate.
+ * 
+ * @return Returns Status containing a success or a failure with the error message.
+ */
+UStatus UriValidator::validate(const UUri& uri) {
+  UStatus status;
+  if ((nullptr == &uri) || isEmpty(uri)) {
+  status.set_code(UCode::INVALID_ARGUMENT);
+    status.set_message("Uri is empty.");
+    return status;
+  }
+  if (uri.has_authority() && !isRemote(uri.authority())) {
+    status.set_code(UCode::INVALID_ARGUMENT);
+    status.set_message("Uri is remote missing uAuthority.");
+    return status;
+  }
+  if (!isEmpty(uri.entity())  && isBlank( uri.entity().name())) {
+    status.set_code(UCode::INVALID_ARGUMENT);
+    status.set_message("Uri is missing uSoftware Entity name.");
+    return status;
+  }
+  status.set_code(UCode::OK);
+  status.set_message("validate Success.");
+  return status;
+}
+
+/**
+ * Validate a {@link UUri} that is meant to be used as an RPC method URI. Used in Request sink values and Response source values.
+ *
+ * @param uri {@link UUri} to validate.
+ * @return Returns Status containing a success or a failure with the error message.
+ */
+UStatus UriValidator::validateRpcMethod(const UUri& uri) {
+    UStatus status = UriValidator::validate(uri);
+    if (status.code() != UCode::OK) {
+        return status;
+    }
+    if (!UriValidator::isRpcMethod(uri)) {
+      status.set_code(UCode::INVALID_ARGUMENT);
+      status.set_message("Invalid RPC method uri. Uri should be the method to be called, or method from response.");
+      return status;
+    }
+  return status;
+}
+
+/**
+ * Validate a {@link UUri} that is meant to be used as an RPC method URI. Used in Request sink values and Response source values.
+ *
+ * @param uri {@link UUri} to validate.
+ * @return Returns Status containing a success or a failure with the error message.
+ */
+UStatus UriValidator::validateRpcResponse(const UUri& uri) {
+    UStatus status = UriValidator::validate(uri);
+    if (status.code() != UCode::OK) {
+        return status;
+    }
+    if (!UriValidator::isRpcResponse(uri)) {
+        status.set_code(UCode::INVALID_ARGUMENT);
+        status.set_message("Invalid RPC response type.");
+        return status;
+    }
+    
+    return status;
+}
+
+/**
+ * Returns true if URI is of type RPC.
+ *
+ * @param uri {@link UUri} to check if it is of type RPC method
+ * @return Returns true if URI is of type RPC.
+ */
+
+bool UriValidator::isRpcMethod(const UUri& uri) {
+    return (nullptr != &uri && 
```

Comment:
Enclose logical items with their own set of parenthesis. Align with logical operators on the left:
return ((nullptr != &uri)
       && (!isEmpty(uri.resource())
       && (uri.resource().has_instance())
and so on.

Reply:
Done